src/core/tools/common/profiler.cpp

```diff
@@ -114,7 +114,9 @@ namespace dsn {
         {
             uint64_t& qts = task_ext_for_profiler::get(this_);
             uint64_t now = dsn_now_ns();
-            s_spec_profilers[this_->spec().code].ptr[TASK_QUEUEING_TIME_NS]->set(now - qts);
+            auto ptr = s_spec_profilers[this_->spec().code].ptr[TASK_QUEUEING_TIME_NS];
```

Comment:
There is a better way to do this. You only plug the callback into our join-point when this perf-counter is definitely not empty, which has better performance.